src/api-binder/report.ts

```diff
@@ -20,6 +19,13 @@ import { shallowDiff, prune, empty } from '../lib/json';
 
 let lastReport: DeviceState = {};
 let lastReportTime: number = -Infinity;
+// Tracks if unable to report the latest state change event.
+let stateChangeDeferred: boolean = false;
+// How often can we report our state to the server in ms
+const maxReportFrequency = 10 * 1000;
```

Comment:
Going by the comments, I think the vars should be named min intervals and not max frequencies. Readers may misconstrue the semantics.
Happened to bump into this PR while looking at a support ticket. Hope I didn't present as unnecessarily pedantic.

Reply:
Thanks for your sense on frequency vs. interval. I agree that interval describes the case better, but I left it at frequency to stay consistent with existing use and minimize changes.